File: lib/services/auth_service.dart
import 'package:firebase_auth/firebase_auth.dart';

class AuthService {
  final FirebaseAuth _auth = FirebaseAuth.instance;

  // Stream to listen to auth changes (use in provider or riverpod later)
  Stream<User?> get authStateChanges => _auth.authStateChanges();

  // Sign up
  Future<User?> signUp(String email, String password) async {
    try {
      final credential = await _auth.createUserWithEmailAndPassword(
        email: email,
        password: password,
      );
      return credential.user;
    } catch (e) {
      return null;
    }
  }

  // Login
  Future<User?> login(String email, String password) async {
    try {
      final credential = await _auth.signInWithEmailAndPassword(
        email: email,
        password: password,
      );
      return credential.user;
    } catch (e) {
      print('Login error: $e');
      return null;
    }
  }

  // Logout
  Future<void> logout() => _auth.signOut();

  // Current user
  User? get currentUser => _auth.currentUser;
}

File: web/src/services/authService.js
import {
  createUserWithEmailAndPassword,
  signInWithEmailAndPassword,
  signInWithPopup,
  GoogleAuthProvider,
  signOut,
  sendPasswordResetEmail,
  sendEmailVerification,
  updateProfile,
} from 'firebase/auth';
import { auth } from './firebase';

const googleProvider = new GoogleAuthProvider();

// After the user clicks the verification link, Firebase redirects them here.
// window.location.origin = localhost:5173 in dev, your real domain in production.
function verificationSettings() {
  return {
    url: `${window.location.origin}/login`,
    handleCodeInApp: false,
  };
}

export async function signUp(email, password, displayName) {
  const credential = await createUserWithEmailAndPassword(auth, email, password);
  if (displayName) {
    await updateProfile(credential.user, { displayName });
  }
  await sendEmailVerification(credential.user, verificationSettings());
  await signOut(auth);
}

export async function resendVerificationEmail(email, password) {
  const credential = await signInWithEmailAndPassword(auth, email, password);
  if (credential.user.emailVerified) return 'already_verified';
  await sendEmailVerification(credential.user, verificationSettings());
  await signOut(auth);
  return 'sent';
}

export async function checkVerificationStatus(email, password) {
  const { user } = await signInWithEmailAndPassword(auth, email, password);
  if (user.emailVerified) return true;
  await signOut(auth);
  return false;
}

export async function login(email, password) {
  const { user } = await signInWithEmailAndPassword(auth, email, password);
  return user;
}

export async function loginWithGoogle() {
  const credential = await signInWithPopup(auth, googleProvider);
  return credential.user;
}

export async function logout() {
  await signOut(auth);
}

export async function resetPassword(email) {
  await sendPasswordResetEmail(auth, email, {
    url: `${window.location.origin}/login`,
    handleCodeInApp: false,
  });
}

